File: ds-core/src/main/java/eu/groeller/ds/domain/exercise/ExerciseDefinition.java
package eu.groeller.ds.domain.exercise;

import eu.groeller.ds.domain.AbstractEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "exercise_definition",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"name", "type"})
        })
public class ExerciseDefinition extends AbstractEntity {

    @NotNull
    private String name;

    @NotNull
    @Enumerated(EnumType.STRING)
    private ExerciseType type;

}

File: ds-compose/app/src/main/java/eu/groeller/dsui/domain/model/ExerciseDefinition.kt
package eu.groeller.dsui.domain.model

/**
 * Domain model representing the definition of an exercise.
 * This corresponds to the backend's ExerciseDefinition entity.
 */
data class ExerciseDefinition(
    val id: Long,
    val name: String,
    val type: ExerciseTypeEnum
) 